### ISSUE TITLE
[Bug] Main file from url interpreted as html

### ISSUE THREAD

**Issue by Luchillo** [electron/electron#8965]
* Electron version: 1.6.2
* Operating system: Ubuntu 16.10 kernel 4.8.0-41-generic

### Expected behavior

The electron cli should open the main file the same as it would if it were from disk.

### Actual behavior

It does load the file but seems to interpret it as an html or text file.

### How to reproduce

The simplest way is to clone the [electron-quick-start](https://github.com/electron/electron-quick-start), and execute this commands: 
```sh
npm i -g http-server
http-server .
electron http://localhost:8080/main.js
```

Feeling lazy with setting up http? or just want to test in the original repo where the error happens? then clone my repo [Electron-webpack](https://github.com/Luchillo/Electron-Webpack), check develop branch, and run this 2 commands in separate terminals:
```sh
npm start
NODE_ENV=development electron http://localhost:3000/main.js
```

### Aditional info:
I created a Discus topic on this in the electron forum: [[Cli] exec electron with main.js from url](https://discuss.atom.io/t/cli-exec-electron-with-main-js-from-url/40801)

**Comment by MarshallOfSound** [comment]
This is not currently supported and IMO would be ridiculously unsafe it was.  If you want a workflow like this I would suggest using some kind of middleman that downloads the JS script to a temporary location and then launches it.

Just as a "for instance" what would `app.getAppPath()` return if launched from an URL like this.

TLDR: It isn't documented and isn't currently supported

**Comment by Luchillo** [comment]
I guess i'll have to tweak the webpack config to build to disk just the `main.js`.

Thanks, indeed i cannot imagine how code that depends of location (like saving to disk on app subfolder) would work when served like that.
